examples/App/App.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import { Button, StyleSheet, Text, View } from 'react-native';
+import { NativeRouter, Route } from 'react-router-native';
+import Stack from 'react-router-native-stack';
+
+function Header() {
+  return (
+    <View style={styles.header}>
+      <Text style={styles.headerText}>I am a lovely default header</Text>
+    </View>
+  );
+}
+
+function PizzaHeader() {
+  return (
+    <View style={[styles.header, styles.pizzaHeader]}>
+      <Text style={styles.headerText}>I am a PIZZA header</Text>
+    </View>
+  );
+}
+
+function Home({ history }) {
+  return (
+    <View style={styles.screen}>
+      <Text>Home Page</Text>
+
+      <Button title="Pizza Page" onPress={() => history.push('/page/pizza')} />
+
+      <Button title="Taco Page" onPress={() => history.push('/page/taco')} />
+
+      <Button title="Hamburger Page" onPress={() => history.push('/page/hamburger')} />
+
+      <Button title="Fullscreen Page" onPress={() => history.push('/fullscreen')} />
+    </View>
+  );
+}
+
+function Page({ match, history }) {
+  return (
+    <View style={styles.screen}>
+      <Text>You're on a {match.params.name} Page!</Text>
+
+      <Button title="Go Back" color="red" onPress={() => history.goBack()} />
+
+      <Button title="Pizza Page" onPress={() => history.push('/page/pizza')} />
+
+      <Button title="Taco Page" onPress={() => history.push('/page/taco')} />
+
+      <Button title="Hamburger Page" onPress={() => history.push('/page/hamburger')} />
+
+      <Button title="Fullscreen Page" onPress={() => history.push('/fullscreen')} />
+    </View>
+  );
+}
+
+function FullScreen({ history }) {
+  return (
+    <View style={styles.screen}>
+      <Text>I am a FullScreen</Text>
+      <Button title="Go Back" color="red" onPress={() => history.goBack()} />
+    </View>
+  );
+}
+
+export default class App extends Component {
+  render() {
+    return (
+      <NativeRouter>
+        <Stack>
+          <Route exact path="/" component={Home} headerComponent={Header} />
+          <Route exact path="/fullscreen" component={FullScreen} />
+          <Route exact path="/page/pizza" component={Page} headerComponent={PizzaHeader} />
+          <Route path="/page/:name" component={Page} headerComponent={Header} />
```

Comment:
the usage of the `headerComponent` prop and the `Header` and `PizzaHeader` (along with some styling) are basically the only new bits here, everything else I copied from existing `Usage` example app